config/initializers/omniauth.rb

```diff
@@ -1,8 +1,19 @@
 # frozen_string_literal: true
 
-Rails.application.config.middleware.use OmniAuth::Builder do
-  provider :github,
+options = {}.tap do |opts|
+  opts[:scope] = "user:email,repo,delete_repo,admin:org,admin:org_hook"
+
+  if GitHubClassroom.enterprise_instance?
+    hostname = Rails.application.secrets.github_enterprise_hostname
+    opts[:site]          = "https://#{hostname}/api/v3"
+    opts[:authorize_url] = "https://#{hostname}/login/oauth/authorize"
+    opts[:token_url]     = "https://#{hostname}/login/oauth/access_token"
+  end
+
+  Rails.application.config.middleware.use OmniAuth::Builder do
+    provider :github,
     Rails.application.secrets.github_client_id,
     Rails.application.secrets.github_client_secret,
-    scope: "user:email,repo,delete_repo,admin:org,admin:org_hook"
+    { client_options: options }
```

Comment:
This block (starting from `Rails.application.config.middleware.use OmniAuth:: Builder do`) needs to either be moved outside of the `{}.tap` call or the variable passed into `client_options`needs to be `opts` not `options`. Since `options` is initialized after the tap block, this will always pass nil for `client_options`